docs/Install_configure_PostgreSQL_CENTOS7.md

```diff
@@ -0,0 +1,157 @@
+
+Following steps will let you install, configure PostgreSQL server and create a
+`haas` database for development or production version of HaaS on a 
+Centos - 7 server.
+
+For simplicity of configuration and ease of maintenance we will use 
+single name `haas` for creating a system user, database role and database name.
+This is merely a guideline for users new to database setup and administration.
+Experienced users are free to choose any other method that suits their need.
+The end goal is to have a way to have a working PostgreSQL backend for `haas`
+
+## Part 1: Install PostgreSQL server. 
+
+Initialize the system. Configure PostgreSQL to allow password authentication.
+
+**1. Install the requisite packages on your server.**
+
+```
+sudo yum install postgresql-server postgresql-contrib -y
+```
+
+**2. Initialize postgresql:**
+
+```
+sudo postgresql-setup initdb
+```
+
+**3. Replace the term `ident` from following lines**
+in file `/var/lib/pgsql/data/pg_hba.conf` with `md5`
+
+**Before:**
+
+```
+# "krb5", "ident", "peer", "pam", "ldap", "radius" or "cert".  Note that
+host    all             all             127.0.0.1/32            ident
+host    all             all             ::1/128                 ident
+#host    replication     postgres        127.0.0.1/32            ident
+#host    replication     postgres        ::1/128                 ident
+```
+
+**After:**
+
+```
+# "krb5", "ident", "peer", "pam", "ldap", "radius" or "cert".  Note that
+host    all             all             127.0.0.1/32            md5
+host    all             all             ::1/128                 md5
+#host    replication     postgres        127.0.0.1/32            ident
+#host    replication     postgres        ::1/128                 ident
+```
+
+**4. Start postgresql service**
+
+```
+$ sudo systemctl start postgresql
+$ sudo systemctl enable postgresql
+```
+
+##Part 2: Setup system user and database role for HaaS.
+
+Create database named `haas` owned by user also named as `haas`.
+
+**5. Create a system user haas:**
+
+If you are setting up this database for a [production setup](production/INSTALL.rst)
+then you might have already created a system user for managing `HaaS`. In that case skip this 
+step.
+
+If you are setting up database for [development purpose](development/HACKING.rst)
+you will need to create a system user to manage `HaaS` database as follows.
+
+**format:** `useradd <username> -d <path-to-home-dir> -m -r `
+
+In this case both system username and home directory will be named `haas`
+
+```
+useradd haas -d /var/lib/haas -m -r
+```
+
+**6. Create a database role named `haas` with priviledges to:**
+ `-r` create roles
+ `-d` create databases and 
+ `-P` will prompt for the password of the new user. 
+This is necessary since we have configured PostgreSQL to use password authentication.
+
+```
+sudo -i -u postgres
```

Comment:
I think it would be better to create a very limited `haas` DB user that only has access to the `haas` DB.

Giving the user role and DB creation abilities would allow a compromise of the `haas` DB user to enable widespread database privs.